proc/proc.go

```diff
@@ -580,21 +601,26 @@ func initializeDebugProcess(dbp *Process, path string, attach bool) (*Process, e
 		return nil, err
 	}
 
-	if err := dbp.updateThreadList(); err != nil {
-		return nil, err
-	}
-
 	switch runtime.GOARCH {
 	case "amd64":
 		dbp.arch = AMD64Arch()
 	}
 
+	if err := dbp.updateThreadList(); err != nil {
+		return nil, err
+	}
+
 	ver, isextld, err := dbp.getGoInformation()
 	if err != nil {
 		return nil, err
 	}
 
 	dbp.arch.SetGStructOffset(ver, isextld)
+	// SelectedGoroutine can no be set correctly by the call to updateThreadList
```

Comment:
s/can no/can not/
s/without a SetGStructOffset/without calling SetGStructOffset/